strawberryfields/configuration.py

```diff
@@ -160,7 +160,7 @@ def update_config(self):
                 if env in os.environ:
                     # Update from environment variable
                     self._config[section][key] = os.environ[env]
-                elif self._config_file:
+                elif self._config_file and key in self._config_file[section]:
```

Comment:
this feels like something that should lead to a new test maybe?

Reply:
Yes - this line is to fix the `KeyError` issue that we got when we excluded a parameter from the config file (i.e. the file was behaving as though all parameters were required). I can add a test for this.

Reply:
Test added.